src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,192 @@
+
+evaluate(distance::SemiMetric, a::Probabilities, b::Probabilities) = Distances.evaluate(distance, a.p, b.p)
+
+"""
+    StatisticalComplexity <: ComplexityMeasure
+    StatisticalComplexity(est = SymbolicPermutation(),
+                          distance = JSDivergence(),
+                          entropy = Renyi())
+
+Calculate the statistical complexity of a time series `x` as defined by Rosso et al. [^Rosso2007], based on 
+permutation patterns.
+
+## Description
+The statistical complexity ``C_{JS}[P]`` is defined based on the distribution ``P`` of 
+ordinal patterns
+```math
+C_{JS}[P] = Q_J[P, P_e] H_S[P]
+```
+where ``Q_J`` is originally the normalized Jensen-Shannon divergence of the ordinal distribution ``P`` to 
+a uniform distribution ``P_e`` with the same numer of bins, and ``H_S`` is the normalized permutation entropy, 
+`entropy_normalized(x, SymbolicPermutation())`.
+In this implementation, the distance measure defaults to the Jensen-Shannon divergence, but other (semi-)metrics can also be
+chosen. The type of entropy is also a free parameter, but defaults to the (originally used) Renyi entropy.
+
+For an ordinal distribution with a given permutation entropy, there is a minimum and maximum value that the
+statistical complexity can take. The minimum and maximum statistical complexity as a function of the entropy
+is calculated using [`minimum_complexity_entropy`](@ref) and [`maximum_complexity_entropy`](@ref).
+
+Standard usage:
+```julia
+m, τ = 3, 1
+est = SymbolicPermutation(; m, τ)
+c = StatisticalComplexity(; est, entropy=Renyi())
+x = randn(100)
+statistical_complexity = complexity(c, x)
+```
+
+NOTE: I'm not sure if this API is congruent with the general complexity API, but I think
+that, since the "standard" signature is also possible, it should be fine.
+I have implemented the other signature because I find it redundant to calculate the entropy
+twice, which is needed for the complexity anyway and usually calculated in combination with it.
+
+Because the distributions of permutation symbols are needed both for the permutation entropy and the
+statistical complexity, the same can also be achieved using
+```julia
+p = probabilities(est, x)
+permutation_entropy = entropy(est, x)
+statistical_complexity = complexity(c, p; h=permutation_entropy)
+```
+
+[^Rosso2007] Rosso, O. A., Larrondo, H. A., Martin, M. T., Plastino, A., & Fuentes, M. A. (2007). 
+            [Distinguishing noise from chaos](https://doi.org/10.1103/PhysRevLett.99.154102).
+            Physical review letters, 99(15), 154102. 
+"""
+Base.@kwdef struct StatisticalComplexity <: ComplexityMeasure
+    est::ProbabilitiesEstimator = SymbolicPermutation()
+    distance::SemiMetric = JSDivergence()
+    entropy::Entropy = Renyi()
+end
+
+alphabet_length(c::StatisticalComplexity) = alphabet_length(c.est)
+
+function complexity(c::StatisticalComplexity, p::Probabilities; h::Float64) 
```

Comment:
`h` is a ready calculated permutation entropy. I don't think making this a property of `StatisticalComplexity` makes too much sense, because

1. It depends on the ordinal probabilities of the specific dataset that is considered. But we want an instance of  `StatisticalComplexity` to be something we can use for multiple datasets if we keep the parameters, just like the `ProbabilitiesEstimator`, right?
2. I use this here, because, as stated above, the two quantities are usually used on combination. When you calculate the statistical complexity, you need the permutation entropy for it, but you want to know the value of the PE itself anyway, which is why you'll usually already have calculated it. In this process, you have estimated both the probabilities and the PE, which you can then pass to the complexity function to not calculate everything again.

I have been thinking about just writing a function that calculates both quantities (PE and SC) at the same time, and returns a Tuple containing both. Would this be congruent with the complexity API in your opinion, cc @kahaaga? Strictly speaking, the `complexity` function would not really calculate _just_ a complexity then